core/server/worker/src/main/java/alluxio/worker/block/BlockWorkerFactory.java

```diff
@@ -46,29 +51,36 @@ public boolean isEnabled() {
   }
 
   @Override
-  public BlockWorker create(WorkerRegistry registry, UfsManager ufsManager) {
+  public Worker create(WorkerRegistry registry, UfsManager ufsManager) {
     BlockMasterClientPool blockMasterClientPool = new BlockMasterClientPool();
     AtomicReference<Long> workerId = new AtomicReference<>(-1L);
-    BlockStore blockStore;
-    switch (Configuration.global()
-        .getEnum(PropertyKey.WORKER_BLOCK_STORE_TYPE, BlockStoreType.class)) {
-      case PAGE:
-        LOG.info("Creating PagedBlockWorker");
-        blockStore = PagedBlockStore.create(ufsManager, blockMasterClientPool, workerId);
-        break;
-      case FILE:
-        LOG.info("Creating DefaultBlockWorker");
-        blockStore =
-            new MonoBlockStore(new TieredBlockStore(), blockMasterClientPool, ufsManager, workerId);
-        break;
-      default:
-        throw new UnsupportedOperationException("Unsupported block store type.");
+    if (DORA_WORKER_ENABLED) {
+      Worker worker = new PagedDoraWorker(blockMasterClientPool, workerId);
+      registry.add(Worker.class, worker);
```

Comment:
the web server expects a `BlockWorker` instance. we need to relax the requirements there in another PR. https://github.com/Alluxio/alluxio/blob/64648e7db5789b6ca58751c34771e651b373fd71/core/server/worker/src/main/java/alluxio/worker/AlluxioWorkerProcess.java#L112-L115

Reply:
good call, will do